app/controllers/v0/burial_claims_controller.rb

```diff
@@ -7,7 +7,18 @@ class BurialClaimsController < ClaimsBaseController
     service_tag 'burial-application'
 
     def show
-      render_burials_json
+      if (submission_attempt = determine_submission_attempt)
+        state = submission_attempt.aasm_state == 'failure' ? 'failure' : 'success'
+        render(json: { data: { attributes: { state: } } })
+      else
+        render(json: CentralMailSubmission.joins(:central_mail_claim).find_by!(saved_claims: { guid: params[:id] }))
+      end
+    rescue ActiveRecord::RecordNotFound => e
+      Rails.logger.error(e.to_s)
+      render(json: { data: { attributes: { state: 'not found' } } }, status: :not_found)
+    rescue => e
+      Rails.logger.error(e.to_s)
+      render(json: { data: { attributes: { state: 'error processing request' } } }, status: :unprocessable_entity)
     end
```

Comment:
It's usually a bad practice to do an assignment in a conditional and to throw an exception in this case. You could do this instead:

```ruby
def show
  if last_submission_attempt
    state = last_submission_attempt.aasm_state == 'failure' ? 'failure' : 'success'
    render(json: { data: { attributes: { state: } } })
  elsif central_mail_submission
    render(json: central_mail_submission)
  else
    render(json: { data: { attributes: { state: 'not found' } } }, status: :not_found)
  end
end

private

def central_mail_submission
  CentralMailSubmission.joins(:central_mail_claim).find_by(saved_claims: { guid: params[:id] })
end

def last_submission_attempt
  claim = claim_class.find_by(guid: params[:id])
  form_submission = claim&.form_submissions&.last
  form_submission&.form_submission_attempts&.last
end
```

Reply:
You're right  on that conditional, but I've always been partial to raising errors when we need to track it in the logs.  I can just log a message instead in this case